### PR TITLE
Unzip ais

### PR DESCRIPTION
**Pull request by MorganMasters**

### CODE REVIEW

**Comment by Jacob-Stevens-Haas** [comment]
Confused by why you merged `download-ais` into this branch.  As I understand it, this PR is really just 32043e4 and 3189244.

Try checking out this branch and running `git reset 3189244` to undo the merge.  Then you will have to force push, `git push -f`.

**Comment by Jacob-Stevens-Haas** [comment]
Ok good, that reset helps.  Now, I know we're talking about wget and other stuff on another branch, and the first commit here isn't really germane to this unzipping function.  So we can use the interactive rebase to remove that commit.

From this branch, try 'git rebase -i HEAD~4'.

You will get a text editor with the list of commits preceded by "pick".  You can then remove the spurious commit from this branch by changing "pick" to "drop" in front.  When you save and quit the editor, git rewinds the branch, then reapplies the diffs for each commit in the appropriate order.  

It's a powerful tool to reorder commits, combine "Work in progress" commits, and remover certain commits from branches. 💪

**Comment by Jacob-Stevens-Haas** [comment]
Ok things looked weird, so I fixed them.  For some reason you had two commits labeled "commit post reset", but idk what they were supposed to accomplish.  So I removed them.

File: tehom/downloads.py [hunk 1]
```diff
@@ -94,8 +95,9 @@ def _unzip_ais(zipfile: Path) -> Tuple[Path]:
         tuple comprising the root of the unzip tree and the specific
         unzipped file of interest
     """
-    # morgan
-    pass
+    with ZipFile(f"{zipfile}", "r") as zipObj:
+        zipObj.extractall(_persistence.AIS_TEMP_DIR)
+    return _persistence.AIS_TEMP_DIR, zipfile.parts[-1]
```
Comment (Jacob-Stevens-Haas) [inline tehom/downloads.py]: First return value: We want to return the root of the unzip tree, not the folder that it's in.

Second return value: `zipfile` is a path to the zipped file.  We want to return the unzipped file's path.

**Review by Jacob-Stevens-Haas — CHANGES_REQUESTED** [review]
A slight difference with return values.  Also, can you write a test for this?

**Comment by MorganMasters** [comment]
By the root of the unzipped tree, do you mean the directory containing the
unzip file's directory?

File: tehom/downloads.py (hunk 1 above)
Comment (MorganMasters) [reply tehom/downloads.py]: how is this?

with ZipFile(f"{zipfile}", "r") as zipObj:
      zipObj.extractall(_persistence.AIS_TEMP_DIR)
  return _persistence.STORAGE, _persistence.AIS_TEMP_DIR / zipfile.parts[-1]

**Comment by MorganMasters** [comment]
Changed to:

+    with ZipFile(f"{zipfile}", "r") as zipObj:
+        zipObj.extractall(_persistence.AIS_TEMP_DIR)
+    return _persistence.STORAGE, _persistence.AIS_TEMP_DIR / zipfile.parts[-1]

Is this what you mean?

**Comment by MorganMasters** [comment]
ah, wait, I see an issue

fixing

File: tehom/downloads.py (hunk 1 above)
Comment (MorganMasters) [reply tehom/downloads.py]: ^ ignore that

**Comment by Jacob-Stevens-Haas** [comment]
First off, your `git pull` merged the remote branch (github branch) with your local branch, reintroducing all the commits I removed.  On your local branch, run `git reset --hard 813cafb`.  Then `git push -f`

I also may have started some confusion; we need to stay using 2015 data and later, since data wasn't kept in csv format before 2015.  Using the example of January 2016, zone 1:
Downloaded filename: `zipfile=PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_2016_01_Zone01.zip')`
When unzipped in place, you now have a file `../ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv`
I would like the return values to be:
```
PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month'), PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv)
```
The reason is because the calling script needs to a) know the path to the csv in order to load it and (b), if successful, delete the unzipped directory.  Currently, the return values are 
```
PosixPath('/'), PosixPath('/home/xenophon/github/tehom/tehom/storage/ais/AIS_ASCII_by_UTM_Month/2016/AIS_2016_01_Zone01.csv')
```

**Review by Jacob-Stevens-Haas — APPROVED** [review]

**Pull request merged**